File: java/whatdoesthefoxsay.java
import java.util.Scanner;

public class whatdoesthefoxsay {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int T = scan.nextInt();
		scan.nextLine();
		while (T-- > 0) {
			String[] sounds = scan.nextLine().split(" ");
			String animal_sound;
			while (!(animal_sound = scan.nextLine()).equals("what does the fox say?")) {
				animal_sound = animal_sound.split(" ")[2];
				for(int i = 0; i < sounds.length; i++)
					if(sounds[i].equals(animal_sound))
						sounds[i] = "";
			}
			for(String s : sounds)
				if(!s.isEmpty())
					System.out.print(s + " ");
			System.out.println();
		}
		scan.close();
	}
}

File: ts/src/whatdoesthefoxsay.ts
import readline = require('readline')

function getFoxSaid (sounds: string[], animalSounds: string[]): string[] {
  animalSounds.forEach(sound => {
    sounds = sounds.filter(s => s !== sound)
  })
  return sounds
}

let sounds: string[] = []
let animalSounds: string[] = []
let lineCount = 0

readline
  .createInterface({
    input: process.stdin,
    output: process.stdout
  })
  .on('line', (line: string) => {
    if (lineCount > 0) {
      if (line === 'what does the fox say?') {
        console.log(getFoxSaid(sounds, animalSounds).join(' '))
        sounds = []
        animalSounds = []
      } else {
        const lineArray = line.split(' ')
        if (lineArray.length === 3 && lineArray[1] === 'goes') {
          animalSounds.push(lineArray[2] ?? '')
        } else {
          sounds.push(...lineArray)
        }
      }
    }
    lineCount++
  })
  .on('close', () => {
    process.exit(0)
  })
